lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1417,26 +1417,32 @@ sub register_assets_from_settings {
 
     # check assets and fix the file names
     for my $k (keys %assets) {
-        my $a = $assets{$k};
-        if ($a->{name} =~ /\//) {
-            log_info "not registering asset $a->{name} containing /";
+        my $asset = $assets{$k};
+        my ($name, $type) = ($asset->{name}, $asset->{type});
+        unless ($name && $type) {
+            log_info 'not registering asset with empty name or type';
             delete $assets{$k};
             next;
         }
-        my $f_asset = _asset_find($a->{name}, $a->{type}, \@parents);
+        if ($name =~ /\//) {
+            log_info "not registering asset $name containing /";
```

Comment:
I would change this to error too. While it's not fatal, it's something to keep an eye on...

Reply:
If it's an error, it should be reported to the client - and raise an exception. Virtualization uses REPO_0=fixed/SLE... because they use REPO_0 in the pxe bootloader

Reply:
I can try to propagate this error higher in the stack. That would be a little bit more work because it looks like the method is called 2 times and each not that "close to the user".

Reply:
Agree with @coolo, this is input validation and failures should be reported to the client instead of getting logged.

Reply:
Ok, I'll change this. But for me this is not a reason to postpone merging this because we had this logging before, too.

Reply:
Agreed on that too @Martchus. Might be worth revisiting input validation in general, since pretty much none of the API controllers do any validation themselves and leave it all to the model layer. Which is quite risky.